bolt-simple-taproot.md

```diff
@@ -0,0 +1,943 @@
+broadcaster can sweep their funds after a delay.  The control block to spend is
+only `33` bytes, as it just includes the internal key (along with the y-parity
+bit and leaf version):
+```
+delay_control_block = (parity_of_y | 0x0c) || <revocationpubkey>
+```
+
+A valid witness is then:
+```
+<delay_control_block> <to_delay_script> <local_delayedsig>
+```
+
+A with base channels, the `nSequence` field must be set to `to_self_delay`.
+
+#### To Remote Outputs
+
+As we inheirit the anchor output semmeantics we want to ensure that the remote
+party can unilaterally sweep their funds after the 1 block CSV delay. In order
+to achieve this property, we'll utilize a NUMs point (`simple_taproot_nums`) By
+using this point as the internal key, we ensure that the remote party isn't
+able to by pass the CSV delay.
+
+The to remote output has the following form:
+
+  * `OP_1 to_remote_output_key`
+  * where:
+    * `taproot_nums_point = 0245b18183a06ee58228f07d9716f0f121cd194e4d924b037522503a7160432f15`
+    * `to_remote_output_key = taproot_nums_point + tagged_hash("TapTweak", taproot_nums_point|| to_remote_script_root`
+    * `to_remote_script_root = tapscript_root([to_remote_script])`
+    * `to_remote_script` is the remote script:
+        ```
+        <remotepubkey> OP_CHECKSIGVERIFY 1 OP_CHECKSEQUENCEVERIFY
+        ```
+
+This output can be swept by the remote party with the following witness:
+```
+<to_remote_control_block> <to_remote_script> <remote_sig>
+```
+
+where `to_remote_control_block` is:
+```
+(parity_of_y(remotepubkey) | 0x0c) || <remotepubkey>
+```
+
+#### Anchor Outputs
+
+For simple taproot channels (`option_simple_taproot`) we'll maintain the same
+form as base segwit channels, but instead will utilize `local_delayedpubkey`
+and the `remotepubkey` rather than the multi-sig keys as that's no longer
+revealed due to musig2.
+
+An anchor output has the following form:
+
+  * `OP_1 anchor_output_key`
+  * where:
+    * `anchor_internal_key = remotepubkey/local_delayedpubkey`
+    * `anchor_output_key= anchor_internal_key + tagged_hash("TapTweak", anchor_internal_key || anchor_script_root)`
+    * `anchor_script = tapscript_root([anchor_script])`
+    * `anchor_script`:
+        ```
+        OP_16 OP_CHECKSEQUENCEVERIFY
+        ```
+
+### HTLC Scripts & Transactions
+
+#### Offered HTLCs
+
+We retain the same second-level HTLC mappings as base channels. The main
+modifications are using the revocation public key as the internal key, and
+splitting the timeout and success paths into individual script leaves.
+
+An offered HTLC has the following form:
+
+  * `OP_1 offered_htlc_key`
+  * where:
+    * `offered_htlc_key = revocation_pubkey + tagged_hash("TapTweak", revocation_pubkey || htlc_script_root)`
+    * `htlc_script_root = tapscript_root([htlc_timeout, htlc_success])`
+    * `htlc_timeout`:
+        ```
+        <local_htlcpubkey> OP_CHECKSIGADD <remote_htlcpubkey> OP_CHECKSIGADD 2 OP_EQUALVERIFY
+        1 OP_CHECKSEQUENCEVERIFY
```

Comment:
`OP_CHECKSIGADD` is added as a replacement for t of n multisignatures where t is not n, simple n of n multisignatures can use `<P1> OP_CHECKSIGVERIFY <P2> OP_CHECKSIGVERIFY ... <Pn> OP_CHECKSIG` (checksig only for the last one)

btw this script is anyone can spend with `<> <> <2>`

https://github.com/Roasbeef/lightning-rfc/pull/1

Reply:
Oh shit, nice catch - these should be `OP_CHECKSIGVERIFY` then @Roasbeef